### DISCUSSION TITLE
Hallucinations different model sizes

### DISCUSSION THREAD

**Discussion by RikRaesTNO** [openai/whisper#1452] [Q&A]
I used Whisper to transcribe the Common Voice data set for a language and note that the 'tiny' model hallucinates a lot, whereas the bigger 'small' model almost does not hallucinate at all, and the even bigger 'base' model hallucinates more than the 'small' model. Furthermore, the general performance of the small model is better than both the tiny and base models. As a side note, the data instances in this data set are sentences worth about 5-10 seconds of audio.

I am mostly interested in your thoughts on why a larger model does not necessarily perform better and may hallucinate more. I did not change any of the temperature or other settings when transcribing. I can imagine a larger model might overfit which can cause this phenomenon but I would like to know what you guys think might be the cause of the lower performance with more hallucinations.

As context: I am doing research for my master thesis so any ideas are welcome!

**Comment by phineas-pta** [comment]
one of the main causes of hallucinations should be training data, not model size

also there're various settings for whisper, you cannot know how those settings + model size would affect hallucinations, default values wouldn't always guarantee best transcripts

if you have time + resources, try changing those settings + data (amount of silence) to compare the performance, also testing pre-process audio, that would be better addition to your thesis

**Comment by jwnacnud** [comment]
In my testing I have found that most hallucinations occur after silence,
thus using ASD to remove the silence does a great job of removing the
hallucinations.

Jeffrey Duncan

**Reply by phineas-pta** [reply]
what is ASD ?

**Comment by jwnacnud** [comment]
Whoops, I meant VAD (Voice Activity Detector) - you can use it to find the
areas of a recording that are silent and trim them out:

https://github.com/snakers4/silero-vad

**Comment by RikRaesTNO** [comment]
I agree with pre-processing the data @phineas-pta , would you have other additional pre-processing steps next to removing silences?

**Reply by phineas-pta** [reply]
that's all i know

**Comment by EtienneAb3d** [comment]
> I agree with pre-processing the data @phineas-pta , would you have other additional pre-processing steps next to removing silences?

You should have a look here:
https://github.com/openai/whisper/discussions/679

And here:
https://github.com/EtienneAb3d/WhisperHallu/tree/main

;)

**Comment by LaurinmyReha** [comment]
maybe this is useful for someone here still:
I wrote a article about this
https://nyra-labs.com/research/killing-hallucinations
which improves this behaviour quite drastically. Feel free to try out the model too :)
https://huggingface.co/nyralabs/CrisperWhisper2.0_large
